classes/that.sc

```diff
@@ -21,51 +21,54 @@ That {
 		all = ();
 	}
 
-	*new {|name, input, analyzerFunction, callback|
+	*new { |name, input, analyzerFunction, callback|
 		var res = all.at(name);
 		if(res.isNil, {
-			if(analyzerFunction.isNil, {
-				Error("Please provide an analyzerFunction").throw;
-			});
-			callback = callback ? {}; // enforce a function as callback and not allow nil
-			res = super.newCopyArgs(name, input, analyzerFunction, callback).init;
```

Comment:
as we do not use `newCopyArgs` anymore: where do we attach the name property to the instance?

Reply:
`newCopyArgs` is just a convenience (and it can be more efficient if you initialise thousands of objects). We just set the variables via the `setFunctions` call, like any outsider could do.